File: Shared/Resources/bogosort.bundle/bogosort.cs
using System;

public class BogoSort {
  public static Random r = new Random();
  
  public static bool IsSorted(int[] arr) {
    for (int i = 1; i < arr.Length; i++) {
      if (arr[i - 1] > arr[i]) {
        return false;
      }
    }
    return true;
  }
  
  public static void Shuffle(int[] arr) {
    int n = arr.Length;
    for (int i = 0; i < n; i++) {
      int a = r.Next(n);
      int b = r.Next(n);
      (arr[a], arr[b]) = (arr[b], arr[a]);
    }
  }
  
  public static void Sort(int[] arr) {
    int n = arr.Length;
    while (!IsSorted(arr)) {
      Shuffle(arr);
    }
  }

  public static void Main(String[] args) {
    int[] array = {0, 39, 21, 62, 91, 77, 14, 23};
    Sort(array);
    string result = "[" + String.Join(", ", array) + "]";
    Console.WriteLine(result);
  }
}

File: Shared/Resources/bogosort.bundle/bogosort.c
#include <stdio.h>
#include <stdlib.h>

int array[8] = {0, 39, 21, 62, 91, 77, 14, 23};

void swap(int *a, int *b) {
  int t = *a;
  *a = *b;
  *b = t;
}

void printList(int items[], int size) {
  for (int i = 0; i < size; i++) {
    if (i == 0) {
      printf("[%d, ", items[i]);
    } else if (i != size - 1) {
      printf("%d, ", items[i]);
    } else {
      printf("%d]", items[i]);
    }
  }
}

int isSorted(int arr[], int n) {
  while (--n >= 1) {
    if (arr[n] < arr[n - 1]) {
      return 0;
    }
  }
  return 1;
}

void sort(int arr[], int n) {
  while (!isSorted(arr, n)) {
    for (int i = 0; i < n; i++) {
      int r = rand() % n;
      swap(&arr[i], &arr[r]);
    }
  }
}

int main(int argc, char *argv[]) {
  int size = sizeof(array) / sizeof(array[0]);
  sort(array, size);
  printList(array, size);
  return 0;
}
